subsys/random/rand32_timer.c

```diff
@@ -14,21 +14,38 @@
  * provide a random number generator.
  */
 
+#include <zephyr/init.h>
 #include <zephyr/random/rand32.h>
 #include <zephyr/drivers/timer/system_timer.h>
 #include <zephyr/kernel.h>
-#include <zephyr/sys/atomic.h>
 #include <string.h>
+#include <zephyr/device.h>
 
 #if defined(__GNUC__)
 
-/*
- * Symbols used to ensure a rapid series of calls to random number generator
- * return different values.
- */
-static atomic_val_t _rand32_counter;
 
-#define _RAND32_INC 1000000003U
+static uint32_t state[4] = {0, 0x9b64c2b0, 0x86d3d2d4, 0xa00ae278};
```

Comment:
Probably good to add a comment about where these magic numbers come from.

Reply:
@galak, these are the same magic numbers found in xoshiro.c and from the comment they are just some pseudo random values